recipes/librasterlite/all/patches/0002-nmake-honor-flags.patch

```diff
@@ -0,0 +1,43 @@
+This patch for msvc build allows to:
+* define OPTFLAG ourself from conanfile (allow to honor profile)
+* inject a custom SYSTEM_LIBS variable from conanfile (allow robust shared build)
+* not hardcode this very specific C:\OSGeo4W environment
+* define rasterliteGetVersion() to be compliant with autotools build
+
+--- a/makefile.vc
++++ b/makefile.vc
+@@ -2,7 +2,6 @@
+ #
+ # NMAKE Makefile to build librasterlite on Windows
+ #
+-!INCLUDE nmake.opt
+ 
+ OBJ_EXT = obj
+ EXT = $(OBJ_EXT)
+@@ -11,10 +10,10 @@ LIBOBJ = lib\rasterlite.$(EXT) lib\rasterlite_gif.$(EXT) \
+ 	lib\rasterlite_png.$(EXT) lib\rasterlite_jpeg.$(EXT) \
+ 	lib\rasterlite_io.$(EXT) lib\rasterlite_image.$(EXT) \
+ 	lib\rasterlite_tiff.$(EXT) lib\rasterlite_aux.$(EXT) \
+-	lib\rasterlite_quantize.$(EXT)
++	lib\rasterlite_quantize.$(EXT) lib\rasterlite_version.$(EXT)
+ RASTERLITE_DLL 	       =	rasterlite$(VERSION).dll
+ 
+-CFLAGS	=	/nologo -IC:\OSGeo4W\include -I.\headers $(OPTFLAGS)
++CFLAGS	=	/nologo -I.\headers $(CFLAGS) $(OPTFLAGS)
+ 
+ CFLAGS2	=	/nologo -IC:\OSGeo4W\include -I.\headers $(OPTFLAGS2)
+ 
+@@ -30,11 +29,8 @@ rasterlite.lib:	$(LIBOBJ)
+ $(RASTERLITE_DLL):	rasterlite_i.lib
+ 
+ rasterlite_i.lib:	$(LIBOBJ)
+-	link /debug /dll /out:$(RASTERLITE_DLL) \
+-		/implib:rasterlite_i.lib $(LIBOBJ) \
+-		C:\OSGeo4W\lib\jpeg_i.lib C:\OSGeo4W\lib\libtiff_i.lib \
+-		C:\OSGeo4W\lib\libpng13.lib C:\OSGeo4W\lib\zlib.lib \
+-		C:\OSGeo4W\lib\geotiff_i.lib C:\OSGeo4W\lib\spatialite_i.lib
++	link /dll /out:$(RASTERLITE_DLL) \
++		/implib:rasterlite_i.lib $(LIBOBJ) $(SYSTEM_LIBS)
```

Comment:
In the recipe, you set `SYSTEM_LIBS` to the system libraries of the dependencies.
How are the libraries of the dependencies themselves linked? Or aren't they required?

Are the libraries of the dependencies included in `self.deps_cpp_info.system_libs`?

Reply:
It's automatically handled by `VisualStudioBuildEnvironment`: https://docs.conan.io/en/latest/reference/build_helpers/visual_studio.html#visualstudiobuildenvironment

Reply:
Are you sure?
The documentation says it sets the `CL` and `LIB` environment variable.
So does it add the the libraries of the dependencies to the `CL` variable?

Shouldn't it be considered a bug that system libraries are not added?

Reply:
I checked the logs and it contains:
```
	link /dll /out:rasterlite.dll  /implib:rasterlite_i.lib lib\rasterlite.obj lib\rasterlite_gif.obj  lib\rasterlite_png.obj lib\rasterlite_jpeg.obj  lib\rasterlite_io.obj lib\rasterlite_image.obj  lib\rasterlite_tiff.obj lib\rasterlite_aux.obj  lib\rasterlite_quantize.obj lib\rasterlite_version.obj shell32.lib Ole32.lib wsock32.lib crypt32.lib ws2_32.lib advapi32.lib user32.lib
Microsoft (R) Incremental Linker Version 14.28.29333.0
Copyright (C) Microsoft Corporation.  All rights reserved.

geotiff.lib libpng16.lib spatialite.lib proj.lib freexl.lib rttopo.lib libxml2_a.lib minizip.lib tiffxx.lib tiff.lib sqlite3.lib libcurl.lib iconv.lib charset.lib geos_c.lib geos.lib bz2.lib zlib.lib libdeflatestatic.lib lzma.lib libjpeg.lib jbig.lib zstd_static.lib webpdecoder.lib webpdemux.lib webpmux.lib webp.lib libssl.lib libcrypto.lib 
   Creating library rasterlite_i.lib and object rasterlite_i.exp
```
So it looks like it does.